components/org.wso2.carbon.streaming.integrator.core/src/main/java/org/wso2/carbon/streaming/integrator/core/internal/ServiceComponent.java

```diff
@@ -163,36 +160,40 @@ protected void start(BundleContext bundleContext) throws Exception {
         StreamProcessorDataHolder.setSiddhiManager(siddhiManager);
         StreamProcessorDataHolder.setStatisticsConfiguration(statisticsConfiguration);
 
-        File runningFile;
+        File siddhiAppFileReference;
 
-        if (runningFileName != null) {
-            StreamProcessorDataHolder.getInstance().setRuntimeMode(SiddhiAppProcessorConstants.RuntimeMode.RUN_FILE);
-            if (runningFileName.trim().equals("")) {
+        if (siddhiAppsReference != null) {
+            if (siddhiAppsReference.trim().equals("")) {
                 // Can't Continue. We shouldn't be here. that means there is a bug in the startup script.
                 log.error("Error: Can't get target file to run. System property {} is not set.",
```

Comment:
Are we targetting one file here? If so, shall we rename `siddhiAppsReference` to `siddhiAppReference`?

Reply:
This will be a folder with several siddhi apps.

Reply:
Ok.... Thought so by reading the error log.